assets/js/googlesitekit/data/create-settings-store.test.js

```diff
@@ -0,0 +1,453 @@
+
+				// True after updating settings on client.
+				dispatch.setSettings( clientValues );
+				expect( select.haveSettingsChanged() ).toEqual( true );
+
+				// False after updating settings back to original server value on client.
+				dispatch.setSettings( serverValues );
+				expect( select.haveSettingsChanged() ).toEqual( false );
+			} );
+		} );
+
+		// Tests for "pseudo-selector" getSetting, available via setting-specific "get{SettingSlug}".
+		describe( 'getSetting', () => {
+			it( 'has the correct selector name', () => {
+				expect( Object.keys( storeDefinition.selectors ) ).toEqual(
+					expect.arrayContaining( [ SETTING.selector ] )
+				);
+			} );
+
+			it( 'uses a resolver to make a network request', async () => {
+				const value = 'serverside';
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/
+					)
+					.mockResponseOnce(
+						JSON.stringify( {
+							otherSetting: 'other-value',
+							[ SETTING.slug ]: value,
+						} ),
+						{ status: 200 }
+					);
+
+				// Setting will have its initial value while being fetched.
+				expect( select[ SETTING.selector ]() ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						select.getSettings() !== undefined
+					),
+				);
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( select[ SETTING.selector ]() ).toEqual( value );
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+			} );
+		} );
+	} );
+
+	describe( 'controls', () => {
+		describe( 'FETCH_SETTINGS', () => {
+			it( 'requests from the correct API endpoint', async () => {
+				const [ type, identifier, datapoint ] = STORE_ARGS;
+				const response = { type, identifier, datapoint };
+
+				fetch
+					.mockResponseOnce( async ( req ) => {
+						if ( req.url.startsWith( `/google-site-kit/v1/${ type }/${ identifier }/data/${ datapoint }` ) ) {
+							return Promise.resolve( {
+								body: JSON.stringify( response ),
+								init: { status: 200 },
+							} );
+						}
+						return Promise.resolve( {
+							body: JSON.stringify( {
+								code: 'incorrect_api_endpoint',
+								message: 'Incorrect API endpoint',
+								data: { status: 400 },
+							} ),
+							init: { status: 400 },
+						} );
+					} );
+
+				const result = await storeDefinition.controls.FETCH_SETTINGS();
+				expect( result ).toEqual( response );
+				// Ensure `console.error()` wasn't called, which will happen if the API
+				// request fails.
+				expect( global.console.error ).not.toHaveBeenCalled();
+			} );
+		} );
+
+		describe( 'FETCH_SAVE_SETTINGS', () => {
+			it( 'requests from the correct API endpoint', async () => {
+				const [ type, identifier, datapoint ] = STORE_ARGS;
+				const response = { type, identifier, datapoint };
+
+				fetch
+					.mockResponseOnce( async ( req ) => {
+						if ( req.url.startsWith( `/google-site-kit/v1/${ type }/${ identifier }/data/${ datapoint }` ) ) {
+							return Promise.resolve( {
```

Comment:
Since this is an async function already, I think you should be able to just return the object here without the `Promise.resolve()` bit. Not a biggie, but might clean things up a bit 🤷‍♀